modules/scatteredBidAdapter.js

```diff
@@ -0,0 +1,175 @@
+// jshint esversion: 6, es3: false, node: true
+'use strict';
+
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { mergeDeep, deepAccess, parseSizesInput, parseUrl, deepSetValue, logInfo, createTrackPixelHtml } from '../src/utils.js';
+
+const { getConfig } = config;
+
+const BIDDER_CODE = 'scattered';
+
+export function buildImps(validBidRequests) {
+  return validBidRequests.map((bid, index) => {
+    let imp = {
+      id: `${index + 1}`
+    };
+    const bannerParams = deepAccess(bid, 'mediaTypes.banner');
+    if (bannerParams && bannerParams.sizes) {
+      const sizes = parseSizesInput(bannerParams.sizes);
+      const format = sizes.map(size => {
+        const [width, height] = size.split('x');
+        const w = parseInt(width, 10);
+        const h = parseInt(height, 10);
+        return { w, h };
+      });
+
+      imp.banner = {
+        format
+      };
+    };
+    return imp;
+  });
+}
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [BANNER],
+
+  // 1.
+  isBidRequestValid: function (bid) {
+    const bidderDomain = deepAccess(bid, 'params.bidderDomain')
+    if (bidderDomain === undefined || bidderDomain === '') {
+      return false
+    }
+
+    const sizes = deepAccess(bid, 'mediaTypes.banner.sizes')
+    if (sizes === undefined || sizes.length < 1) {
+      return false
+    }
+
+    return true
+  },
+
+  // 2.
+  buildRequests: function (validBidRequests, bidderRequest) {
+    const ortb2Obj = bidderRequest.ortb2 || {};
+    let { user } = ortb2Obj
+
+    let site = getConfig('site') || {};
```

Comment:
For a new adapter I believe it's better to not do this, since this way of setting FPD is deprecated - and incomplete (the same should be done for `app` and `device`). We have [this issue](https://github.com/prebid/Prebid.js/issues/9335) open to solve this problem in core, after which adapters would see everything directly under `bidderRequest.ortb2`.